TiDB Cloud Docs: Add documentation for new Recovery Group feature (#17425)
pingcap/docs: tidb-cloud/recovery-group-get-started.md

---
title: Get Started with Recovery Groups
summary: Learn how to create a Recovery Group in TiDB Cloud and view its details
---

# Get Started with Recovery Groups

This document describes how to create a Recovery Group to protect your databases running on TiDB Cloud Dedicated Clusters using the TiDB Cloud user interface. It also shows how to view details of the recovery group.

> ## Prerequisites

Reviewer: After Recovery Group is created, we will create account(in this pattern cloud-rg-*) in the secondary cluster, which will be used to replicated data into secondary cluster, we should remind user not to touch such account. If mistakenly delete such account, it will cause replication interrupted. 
I not should where to put this remind message, just comment here.

Recovery Groups replicate your databases to another cluster to protect your databases from a regional disaster. To create a recovery group you will need to have two TiDB Cloud Dedicated Clusters, one where the databases are primarily located, and a second cluster that will be used for the replicated copy of the databases. If you have not done so already, follow the steps in [Create a TiDB Dedicated Cluster](/tidb-cloud/create-tidb-cluster.md) to create the necessary clusters.

> **Note**
>
> TiDB Cloud Recovery Groups are currently supported with TiDB Cloud Dedicated Cluster deployed on AWS

## Create a new Recovery Group

To create a Recovery Group, perform the following steps:

1. In the TiDB Cloud console, click **Project Settings** in the left navigation pane.

2. On the project settings navigation pane, click **Recovery Group**.

3. On the recovery group page, click **Create Recovery Group**. A dialog is displayed.

4. In the Create Recovery Group Dialog enter a name for the recovery group.

    > **Note**
    >

>     > Currently only one resiliency level is supported.

Reviewer: Can you add explanation for what does resiliency level means for the users? What is the expected behavior for the only supported resiliency level?

Reviewer 2: Add a new secton https://github.com/pingcap/docs/pull/17425/files#diff-ff23e50bb2b6fd16bac6cf66da63f4019cc00a42f986fba4ba08be01c0632725R74

5. Select the TiDB Cloud cluster that will be the primary cluster for this group.

6. Select the TiDB Cloud cluster that will be the seondary cluster that databases will be replicated to for this group.

7. Select which databases you wish to replicate as part of this recovery group.

>     > **Note**

Reviewer: We should also remind user here, we are not support replicate system tables to secondary cluster even users select All.

Reviewer: We should also remind users that we need to export a lot data from primary and import to secondary at backend during creation, which may have impact on online query for both primary and secondary cluster. Remind users to perform creation operation when traffic is low.

Author: Addressed in https://github.com/pingcap/docs/pull/17425/commits/da21f7502dd8af02816368bae7e09f501b693fa1

---
title: Get Started with Recovery Groups
summary: Learn how to create a recovery group in TiDB Cloud and view its details.
---

# Get Started with Recovery Groups

This document describes how to create a recovery group to protect your databases running on TiDB Cloud Dedicated clusters using the [TiDB Cloud console](https://tidbcloud.com/). It also shows how to view details of a recovery group.

## Prerequisites

> A recovery group replicates your databases to another cluster to protect your databases from regional disasters. To create a recovery group, you need to have two TiDB Cloud Dedicated clusters. One cluster hosts the primary databases, and a second cluster hosts the replicas of the primary databases. If you have not done so already, follow the steps in [Create a TiDB Dedicated Cluster](/tidb-cloud/create-tidb-cluster.md) to create the necessary clusters.

Reviewer 3: ```suggestion
- A recovery group replicates your databases to another cluster to protect your databases from regional disasters. Before creating a recovery group, you need to have two TiDB Cloud Dedicated clusters. One cluster hosts the primary databases, and a second cluster hosts the replicas of the primary databases. If you have not done so already, follow the steps in [Create a TiDB Dedicated Cluster](/tidb-cloud/create-tidb-cluster.md) to create the necessary clusters.
- To create a recovery group, you must be in the `Organization Owner` role of your organization or the `Project Owner` role of the target project.
```

> 

Reviewer 3: In the table of https://docs.pingcap.com/tidbcloud/manage-user-access#project-roles, a row about managing replication groups needs to be added.

Reviewer 2: https://github.com/pingcap/docs/pull/17425/files#diff-aec01ff042ba824dd4674abecfb9edcd6c4704b481b951a3ad287809ae8908d9

8. Review the summary information, and then click **Create** to begin protecting the databases as part of the new recovery group.

## View recovery group details

After creating a recovery group, you can view its status information on the **Recovery Group Detail** page:

1. In the [TiDB Cloud console](https://tidbcloud.com/), click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner, switch to the target project if you have multiple projects, and then click **Project Settings**.

2. In the **Project Settings** navigation pane, click **Recovery Group**.

3. On the **Recovery Group** page, click the name of the recovery group that you wish to view.

    The **Recovery Group Detail** page provides information about a recovery group, including its configuration details, status, and metrics on the replication throughput and latency. 

4. When a replication relationship is fully established and functioning, the status is displayed as **Available**.

    > **Warning**
    >

>     > During the setup of a recovery group, an account named following the pattern `cloud-rg-* will be created on the secondary cluster for the replication process. Deleting or modifying this account will interrupt the replication.

Reviewer 3: ```suggestion
    > During the setup of a recovery group, an account named following the pattern `cloud-rg-*` will be created on the secondary cluster for the replication process. Deleting or modifying this account will interrupt the replication.
```

- To create a recovery group, you must be in the `Organization Owner` role of your organization or the `Project Owner` role of the target project.

> **Note**
>
> Currently, only TiDB Dedicated clusters hosted on AWS support recovery groups.

## Create a new recovery group

To create a recovery group, perform the following steps:

1. In the [TiDB Cloud console](https://tidbcloud.com/), click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner, switch to the target project if you have multiple projects, and then click **Project Settings**.

2. In the **Project Settings** navigation pane, click **Recovery Group**.

3. On the **Recovery Group** page, click **Create Recovery Group**. 

4. On the **Create Recovery Group** page, enter a name for the recovery group.

    > **Note**
    >

>     > Currently only one resiliency level is supported.

Reviewer 2: ```suggestion
    > Currently only one resiliency level is supported. For more information, see [About resiliency levels](#about-resiliency-levels).
```

## View recovery group details

After creating a recovery group, you can view its status information on the **Recovery Group Detail** page:

1. In the [TiDB Cloud console](https://tidbcloud.com/), click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner, switch to the target project if you have multiple projects, and then click **Project Settings**.

2. In the **Project Settings** navigation pane, click **Recovery Group**.

3. On the **Recovery Group** page, click the name of the recovery group that you wish to view.

    The **Recovery Group Detail** page provides information about a recovery group, including its configuration details, status, and metrics on the replication throughput and latency. 

4. When a replication relationship is fully established and functioning, the status is displayed as **Available**.

    > **Warning**
    >
    > During the setup of a recovery group, an account named following the pattern `cloud-rg-*` will be created on the secondary cluster for the replication process. Deleting or modifying this account will interrupt the replication.

> 

Reviewer 2: ```suggestion

## About resiliency levels

A resiliency level defines the consistency characteristics of data reading in different scenarios of Recovery Group. Currently, TiDB Cloud only provides the following resiliency level:

- No consistency guaranteed. During the replication of a recovery group, the downstream cluster does not guarantee transaction consistency read. When the upstream cluster becomes unavailable, you can not restore the data in the downstream cluster to a transaction consistency state.

TiDB Cloud will provide the other two levels in near future:

- Eventual consistency. During the replication of a recovery group, the downstream cluster does not guarantee transaction consistency read. When the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.
- Near real-time consistency. During the replication of a recovery group, the downstream cluster provides approximate real-time transaction consistency read. When the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.

```

---

# Get Started with Recovery Groups

This document describes how to create a Recovery Group to protect your databases running on TiDB Cloud Dedicated Clusters using the TiDB Cloud user interface. It also shows how to view details of the recovery group.

## Prerequisites

Recovery Groups replicate your databases to another cluster to protect your databases from a regional disaster. To create a recovery group you will need to have two TiDB Cloud Dedicated Clusters, one where the databases are primarily located, and a second cluster that will be used for the replicated copy of the databases. If you have not done so already, follow the steps in [Create a TiDB Dedicated Cluster](/tidb-cloud/create-tidb-cluster.md) to create the necessary clusters.

> **Note**
>
> TiDB Cloud Recovery Groups are currently supported with TiDB Cloud Dedicated Cluster deployed on AWS

## Create a new Recovery Group

To create a Recovery Group, perform the following steps:

1. In the TiDB Cloud console, click **Project Settings** in the left navigation pane.

2. On the project settings navigation pane, click **Recovery Group**.

> 3. On the recovery group page, click **Create Recovery Group**. A dialog is displayed.

Reviewer 2: ```suggestion
3. On the **Recovery Group** page, click **Create Recovery Group**. A dialog is displayed.
```

4. In the Create Recovery Group Dialog enter a name for the recovery group.

    > **Note**
    >
    > Currently only one resiliency level is supported.

5. Select the TiDB Cloud cluster that will be the primary cluster for this group.

6. Select the TiDB Cloud cluster that will be the seondary cluster that databases will be replicated to for this group.

7. Select which databases you wish to replicate as part of this recovery group.

    > **Note**
    >

>     > When assigning databases to the group you can select specific databases to the recovery group, or select all (non-system) databases on the primary cluster (current and future).

Reviewer 2: ```suggestion
    > When assigning databases to the group, you can select specific databases to the recovery group, or select all (non-system) databases on the primary cluster (current and future).
```

    > If you **Assign all databases (current and future)** then any future databases added to the cluster will be automatically included in this recovery group and replicated to the secondary cluster.

>     > If you **Assign specific databases** then you will select the specific databases on the primary cluster that you want to replicate to the secondary cluster. If any databases are added to the primary cluster in the future, then these will not be automatically replicated as part of this recovery group.

Reviewer 2: ```suggestion
    > If you **Assign specific databases** then you will select the specific databases on the primary cluster that you want to replicate to the secondary cluster. If any databases are added to the primary cluster in the future, then these databases will not be automatically replicated as part of this recovery group.
```

    > The page also provides metrics on the throughput of the recovery groups replication, and the replication latency experienced.

4. The status of the recovery group will read as **Available** when the replication relationship is fully established and functioning.

    > **Warning**
    >
    > As part of the setup of a recovery group, an account will be created on the secondary cluster that will be used by the replication process. This account will be named following the pattern `cloud-rg-*`. Deleting or modifying this account will cause the replication to be interrupted.

## What's next

After creating the recovery group, you might want to familiarize yourself with the failover and reprotect operations. These operations are used to **Failover** the primary cluster for the replicated databases from one cluster to the other, and then to later reestablish replication in the opposite direction to **Reprotect** the failed over databases.

- [Failover and Reprotect Databases](/tidb-cloud/recovery-group-failover.md)

> 

Reviewer 2: ```suggestion
```

## View recovery group details

After creating a recovery group, you can view its status information on the **Recovery Group Detail** page:

1. In the [TiDB Cloud console](https://tidbcloud.com/), click <MDSvgIcon name="icon-left-projects" /> in the lower-left corner, switch to the target project if you have multiple projects, and then click **Project Settings**.

2. In the **Project Settings** navigation pane, click **Recovery Group**.

3. On the **Recovery Group** page, click the name of the recovery group that you wish to view.

    The **Recovery Group Detail** page provides information about a recovery group, including its configuration details, status, and metrics on the replication throughput and latency. 

4. When a replication relationship is fully established and functioning, the status is displayed as **Available**.

    > **Warning**
    >
    > During the setup of a recovery group, an account named following the pattern `cloud-rg-*` will be created on the secondary cluster for the replication process. Deleting or modifying this account will interrupt the replication.

## About resiliency levels

> A resiliency level defines the consistency characteristics of data reading in different scenarios of Recovery Group. Currently, TiDB Cloud only provides the following resiliency level:

Reviewer 3: ```suggestion
A resiliency level defines the consistency characteristics of data reading in different scenarios of a recovery group. Currently, TiDB Cloud only provides the following resiliency level:
```

- No consistency guaranteed. During the replication of a recovery group, the downstream cluster does not guarantee transaction consistency read. When the upstream cluster becomes unavailable, you can not restore the data in the downstream cluster to a transaction consistency state.

> TiDB Cloud will provide the other two levels in near future:

Reviewer 3: ```suggestion
TiDB Cloud will provide two additional resiliency levels in the near future:
```

> - Eventual consistency. During the replication of a recovery group, the downstream cluster does not guarantee transaction consistency read. When the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.

Reviewer 3: ```suggestion
- Eventual consistency. During the replication of a recovery group, the downstream cluster does not guarantee transaction consistency read. However, when the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.
```

> - Near real-time consistency. During the replication of a recovery group, the downstream cluster provides approximate real-time transaction consistency read. When the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.

Reviewer 3: ```suggestion
- Near real-time consistency. During the replication of a recovery group, the downstream cluster provides approximately real-time transaction consistency read. When the upstream cluster becomes unavailable, you can restore the data in the downstream cluster to a transaction consistency state.
```
